csp_billing_adapter_microsoft/plugin.py

```diff
@@ -32,16 +32,14 @@
 
 log = logging.getLogger('CSPBillingAdapter')
 
-METADATA_ADDR = 'http://169.254.169.254/metadata'
-METADATA_API_VERSION = '?api-version=2021-12-13'
-METADATA_INSTANCE_PATH = '/instance'
-METADATA_TOKEN_PATH = '/identity/oauth2/token'
-METADATA_TOKEN_RESOURCE = '&resource=https://management.azure.com/'
+METADATA_URL = 'http://169.254.169.254/metadata/'
+INSTANCE_INFO_URL = METADATA_URL + 'instance?api-version={latest}'
+VERSIONS_URL = METADATA_URL + 'versions'
+METADATA_TOKEN = '/identity/oauth2/token?api-version={version}&resource=https://management.azure.com/'
 
-METADATA_INSTANCE_URL = METADATA_ADDR + METADATA_INSTANCE_PATH
-METADATA_TOKEN_URL = \
-    METADATA_ADDR + METADATA_TOKEN_PATH + METADATA_API_VERSION + \
-    METADATA_TOKEN_RESOURCE
+TOKEN_URL = METADATA_URL + METADATA_TOKEN
+SIGNATURE_URL = METADATA_URL + 'attested/document?api-version={latest}'
```

Comment:
This is not a constant , there is a substitution.

WE know the version of the API we want that can be a constant and then this can also become a constant.